src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerWaitContainer.groovy

```diff
@@ -15,19 +15,37 @@
  */
 package com.bmuschko.gradle.docker.tasks.container
 
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+
+import java.util.concurrent.TimeUnit
+
 class DockerWaitContainer extends DockerExistingContainer {
 
     protected int exitCode
 
-    DockerWaitContainer(){
-        ext.getExitCode = {exitCode}
+    /**
+     * Wait timeout in seconds.
+     */
+    @Input
+    @Optional
+    Long timeout
+
+    DockerWaitContainer() {
+        ext.getExitCode = { exitCode }
     }
 
     @Override
     void runRemoteCommand(dockerClient) {
-        logger.quiet "Waiting for container with ID '${getContainerId()}'."
+        logger.quiet "Waiting for container with ID '${getContainerId()}' for ${timeout}s."
         def containerCommand = dockerClient.waitContainerCmd(getContainerId())
-        exitCode = containerCommand.exec()
+        def callback = threadContextClassLoader.createWaitContainerResultCallback()
+        callback = containerCommand.exec(callback)
+        if (timeout) {
+            exitCode = callback.awaitStatusCode(timeout, TimeUnit.SECONDS)
```

Comment:
Lets change TimeUnit to use milliseconds. Anyone familiar with java/groovy, and its various ways to "wait", would expect to pass in milliseconds especially considering we are using a Long (which I completely agree with).